docs/custom_integrations.qmd

```diff
@@ -55,3 +55,28 @@ sections = [
 for section_name, folder_name in sections:
     print(print_section(section_name, folder_name))
 ```
+
+## Adding a new integration
+
+Adding new plugins allows hooking to multiple parts of the training pipeline for custom features.
+
+::: {.callout-note}
+
+See `BasePlugin` at [src/axolotl/integrations/base.py](https://github.com/axolotl-ai-cloud/axolotl/blob/main/src/axolotl/integrations/base.py) for the possible hooks.
+
+:::
+
+To add a new integration, please follow these steps:
+
+1. Create a new folder in the `src/axolotl/integrations` directory.
+2. Add any relevant files (`LICENSE`, `README.md`, `ACKNOWLEDGEMENTS.md`, etc.) to the new folder.
```

Comment:
I am not a copyright expert, but my suggestion here is for those "in-tree" plugins, they should have the same license as the whole axolotl (for simplicity)